### ISSUE TITLE
Expose JSViewCompiler as subspec in Podspec

### ISSUE THREAD

**Issue by adamski** [couchbase/couchbase-lite-ios#1259]
In order to be able to build ReactNativeCouchbaseLite I need to add the CBLJSViewCompiler library. Currently I have to do this by linking libCBLJSViewCompiler.a, and manually adding `CBLRegisterJSViewCompiler.h`. This could be done automatically if JSViewCompiler was exposed as a subspec in the Podspec e.g. 

``` ruby
  s.subspec 'JSViewCompiler' do |ss|
    ss.vendored_library = 'Extras/libCBLJSViewCompiler.a'
    ss.source_files = 'Extras/CBLRegisterJSViewCompiler.h'
    ss.preserve_paths = 'CouchbaseLite.framework'
    ss.vendored_frameworks = 'CouchbaseLite.framework'
  end
```

Or we go with the suggestion in #1257

**Comment by snej** [comment]
@pasin, as the CocoaPods expert ;-) what do you think?

**Comment by snej** [comment]
We could just always include the JSViewCompiler in the library, so a separate build wouldn't be needed. We decided not to do this earlier, but Pasin and I can't remember why — did we not want to make apps link against JavaScriptCore?

**Comment by pasin** [comment]
We added CBLJSViewCompilier into CouchbaseLite (not listener) and rolled that back per https://github.com/couchbase/couchbase-lite-ios/issues/1054#issuecomment-175298309. The main reason that we rolled back because users will need to start adding JavaScriptCore.framework as one of the dependencies.

**Comment by pasin** [comment]
One thing that I can do is to add libCBLJSViewCompiler along with the listener podspec.

**Comment by pasin** [comment]
I added libCBLJSViewCompiler dependency to the podspec. By default libCBLJSViewCompiler as well as JavaScriptCore will be linked. This is a workaround until we move libCBLJSViewCompiler into CouchbaseLite framework. The change has been merged to master and will be available to the cocoapods registry for v1.4.0.
